### ISSUE TITLE
iOS 9.3 testing

### ISSUE THREAD

**Issue by ruslo** [headupinclouds/gatherer#150]
Need to experiment with `osx_image` feature:
- https://docs.travis-ci.com/user/osx-ci-environment/#OS-X-Version

Will it work for Multi-OS configuration?

Block: issue #139

**Comment by ruslo** [comment]
Simple case seems to work:
- https://github.com/travis-ci-tester/travis-test-multi-os-custom-osx-image/blob/master/.travis.yml
- https://travis-ci.org/travis-ci-tester/travis-test-multi-os-custom-osx-image/jobs/133949618

**Comment by ruslo** [comment]
iOS 9.3 testing added to Polly:
- https://travis-ci.org/ruslo/polly/jobs/134211457

**Comment by ruslo** [comment]
I've added iOS 9.3 to GTest testing but found a bug in CMake ios-universal module:
- https://travis-ci.org/ingenue/hunter/jobs/134242681

Need to fix CMake and upload patched version.

**Comment by ruslo** [comment]
> Need to fix CMake and upload patched version

patch sent to upstream: https://cmake.org/pipermail/cmake-developers/2016-June/028681.html
testing experimental version, Polly `develop`: https://travis-ci.org/ruslo/polly/builds/137820318

**Comment by ruslo** [comment]
> testing experimental version, Polly develop

version with fix merged to master, retesting GTest with iOS 9.3:
- https://travis-ci.org/ingenue/hunter/builds/137866445

**Comment by ruslo** [comment]
> retesting GTest with iOS 9.3

cool, it works. will update Gatherer and dependencies

**Comment by ruslo** [comment]
> will update Gatherer and dependencies

Done. Now got error on packing stage (CPack) of iOS 9.3 (build/install works fine)

https://travis-ci.org/headupinclouds/gatherer/builds/138073691

**Comment by ruslo** [comment]
> Now got error on packing stage (CPack)

It might be a nontrivial fix. I've openned an issue for discussion: https://gitlab.kitware.com/cmake/cmake/issues/16149 Will work on packing `_install` directory by Polly.

**Comment by ruslo** [comment]
Not able to build for iOS simulator, i386 build error:

```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
-x c++ -arch i386 -fmessage-length=0 -fdiagnostics-show-note-include-stack
-fmacro-backtrace-limit=0 -Wno-trigraphs -fpascal-strings -O0
-Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type
-Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors
-Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function
-Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value
-Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow
-Wno-four-char-constants -Wno-conversion -Wno-constant-conversion
-Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion
-Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions
-DCMAKE_INTDIR=\"Debug-iphonesimulator\" -DGATHERER_OPENGL_ES
-DGATHERER_ENABLE_OPENGL_DEBUG -DGATHERER_OPENGL_ES=1 -DPNG_DEBUG -isysroot
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.3.sdk
-fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof
-mios-simulator-version-min=9.3 -g -Wno-sign-conversion
-I/Users/travis/build/ruslo/gatherer/_builds/ios-nocodesign-9-3-wo-armv7s/src/lib/Debug-iphonesimulator/include
-I/Users/travis/build/ruslo/gatherer/src/lib
-I/Users/travis/build/ruslo/gatherer/src/3rdparty/ogles_gpgpu
-I/Users/travis/build/ruslo/gatherer/src/3rdparty/ogles_gpgpu/ogles_gpgpu
-I/Users/travis/build/ruslo/gatherer/src/3rdparty/ogles_gpgpu/ogles_gpgpu/common
-I/Users/travis/build/ruslo/gatherer/src/3rdparty/thread-pool-cpp/thread_pool
-I/Users/travis/.hunter/_Base/45ac039/c31b1e8/0b7f9e7/Install/include/opencv
-I/Users/travis/.hunter/_Base/45ac039/c31b1e8/0b7f9e7/Install/include
-I/Users/travis/build/ruslo/gatherer/_builds/ios-nocodesign-9-3-wo-armv7s/src/lib/gatherer.build/Debug-iphonesimulator/gatherer_graphics.build/DerivedSources/i386
-I/Users/travis/build/ruslo/gatherer/_builds/ios-nocodesign-9-3-wo-armv7s/src/lib/gatherer.build/Debug-iphonesimulator/gatherer_graphics.build/DerivedSources
-Wmost -Wno-four-char-constants -Wno-unknown-pragmas
-F/Users/travis/build/ruslo/gatherer/_builds/ios-nocodesign-9-3-wo-armv7s/src/lib/Debug-iphonesimulator
-std=c++11 -MMD -MT dependencies -MF
/Users/travis/build/ruslo/gatherer/_builds/ios-nocodesign-9-3-wo-armv7s/src/lib/gatherer.build/Debug-iphonesimulator/gatherer_graphics.build/Objects-normal/i386/GLExtra.d
--serialize-diagnostics
/Users/travis/build/ruslo/gatherer/_builds/ios-nocodesign-9-3-wo-armv7s/src/lib/gatherer.build/Debug-iphonesimulator/gatherer_graphics.build/Objects-normal/i386/GLExtra.dia
-c /Users/travis/build/ruslo/gatherer/src/lib/graphics/GLExtra.cpp -o
/Users/travis/build/ruslo/gatherer/_builds/ios-nocodesign-9-3-wo-armv7s/src/lib/gatherer.build/Debug-iphonesimulator/gatherer_graphics.build/Objects-normal/i386/GLExtra.o

    In file included from /Users/travis/build/ruslo/gatherer/src/lib/graphics/GLExtra.cpp:9:

    /Users/travis/build/ruslo/gatherer/src/lib/graphics/GLExtra.h:39:22: error: unknown type name 'GLfloat'; did you mean 'float'?

    void glMakeIdentityf(GLfloat m[16]);
                         ^~~~~~~
                         float
```

**Comment by ruslo** [comment]
Fixed in #160
